### ISSUE TITLE
500 Error Setting Up Plugin on Subsite of Multisite Network

### ISSUE THREAD

**Issue by tarecord** [johngodley/redirection#3324]
### Summary

I'm trying to debug an issue related to #3158 on a Multisite install. However, I uninstalled the plugin completely, removed left-over tables ( I had a couple `wp_[site_id]_redirection_xxxx` tables), and reinstalled the plugin. Everything was fine until I tried setting it up on a sub-site.

### Actual Behavior

When I tried activating the plugin on a site that wasn't the main site (ex. `/somesite/` instead of `/`) I was met with a 500 error:
```
PHP Fatal error:  Uncaught Error: Cannot unset string offsets in /application/www/wp-content/plugins/redirection/redirection-settings.php:96
```

I've tracked it down to [red_get_options()](https://github.com/johngodley/redirection/blob/f230ffe8e95e26957ff9228eb1e52be532aa50d1/redirection-settings.php#L314) not checking that the response from `get_option()` is of the correct type. Right now I'm seeing that line is returning `"0"` when it should be an array.

I've seemed to resolve the issue by updating [L314](https://github.com/johngodley/redirection/blob/f230ffe8e95e26957ff9228eb1e52be532aa50d1/redirection-settings.php#L314) to:
```php
$options = (array) get_option( REDIRECTION_OPTION );
```

### Steps to reproduce

1. Install plugin on a multisite.
2. Run through the setup on the main site.
3. Try to run through the setup screen on another subsite on the multisite.

### Environment
**Browser:** Brave
**Browser Version:** [Version 1.39.122 Chromium: 102.0.5005.115 (Official Build) (x86_64)](https://brave.com/latest/)
**Version of WordPress:** 5.8.4
**Redirection Plugin:** 5.2.3

**Comment by johngodley** [comment]
I've not been able to reproduce this problem. I have a multisite with subdirectories (I think that's what you are using), and have create a new site and activated Redirection there. I then went through the Redirection installation process and everything worked fine.

Have you done anything differently?

**Comment by johngodley** [comment]
That's interesting. I'm not sure how a `0` got in there - the option should either exist (and be valid) or not exist at all.

I've released 5.3 beta 2 which includes a modification of your change. Thanks for digging into this further!

https://github.com/johngodley/redirection/releases/tag/5.3-beta

I'll mark this as closed
